guida/src/app/page.tsx

```diff
@@ -1,38 +1,126 @@
+      }
+    });
+    closeModal();
+  };
 
   return (
     <div className={styles.fullScreen}>
       <div className={styles.postContainer}>
         <h1>Feed</h1>
         {error && <div>Error: {error}</div>} {/* Display error if any */}
-        <ul>
-          {loading
-            ? Array.from({ length: 5 }, (_, index) => (
-                <li
-                  key={index}
-                  className={`${styles.skeleton} ${styles["skeleton-post"]}`}
-                ></li> // Render skeleton loaders while loading
-              ))
-            : posts.map(({ id, title, content }) => (
-                <li key={id} className={styles.post}>
-                  <div className={styles.postContent}>
-                    <h2>{title}</h2>
-                    <p>{content}</p>
-                  </div>
-                </li> // Render posts once they are fetched
-              ))}
-        </ul>
+        <button onClick={handleCreatePost}>Create New Post</button> {/* Add button to create new post */}
+        {loading ? (
+          Array.from({ length: 5 }, (_, index) => (
+            <li
+              key={index}
+              className={`${styles.skeleton} ${styles["skeleton-post"]}`}
+            ></li> // Render skeleton loaders while loading
+          ))
+        ) : displayedPosts.length === 0 ? (
+          <div>All posts read</div> // Display message when there are no posts
+        ) : (
+          <ul>
+            {displayedPosts.map(({ id, title, content, published, authorId }) => (
+              <li
+                key={id}
+                className={styles.post}
+                onClick={() => handlePostClick({ id, title, content, published, authorId })} // Pass complete post
+              >
+                <div className={styles.postContent}>
+                  <h2>{title}</h2>
+                  <p>{content}</p>
+                </div>
+              </li> // Render posts once they are fetched
+            ))}
+          </ul>
+        )}
       </div>
+      {(clickedPost || isCreating) && (
+        <Modal onClose={closeModal} 
+        postId={clickedPost?.id ?? 0} 
+        onDelete={handlePostDelete}>
+          {isEditing || isCreating ? (
+            <PostForm
+              post={clickedPost || undefined} // Pass the full clicked post to PostForm
+              submitText={isCreating ? "Create Post" : "Update Post"} // Change submit text for editing
+              onClose={closeModal}
+              onSuccess={handleSuccess} // Close modal on submit
+            />
+          ) : (
+            <>
+              {clickedPost && ( // Add null check before accessing clickedPost properties
+                <>
+                  <h2>{clickedPost.title}</h2>
+                  <p>{clickedPost.content}</p>
+                  <button onClick={handleEditPost}>Edit Post</button> {/* Add Edit button directly */}
+                </>
+              )}            </>
+          )}
+        </Modal>
+      )}
```

Comment:
As in my previous comment I feel like this could be refactored into separate components for both readability and reusability. Instead of passing children to a general component based on state why not render two different relevantly named components based on state?